templates/rust/lib.tera

```diff
@@ -35,18 +36,23 @@ pub use {{module_name}} as csfr_cpu;
 {% endfor -%}
 {% endif %}
 
-
 {% for name,p in ir.peripheral_mod %}
 {%- set module_name = p.name | to_mod_id -%}
+#[derive(Copy, Clone, Eq, PartialEq)] {# Peripheral definition#}
+{% set peri_struct = p.name | to_struct_id -%}
+pub struct {{ peri_struct }}(*mut u8);
+unsafe impl core::marker::Send for {{ peri_struct }} {}
+unsafe impl core::marker::Sync for {{ peri_struct }} {}
```

Comment:
We currently generate e.g.
```
#[derive(Copy, Clone, Eq, PartialEq)]
pub struct DerivedTest(*mut u8);
unsafe impl core::marker::Send for DerivedTest {}
unsafe impl core::marker::Sync for DerivedTest {}
#[cfg(feature = "derivedtest")]
pub const DERIVEDTEST: self::DerivedTest = self::DerivedTest(0xa0000u32 as _);
```

Should we change this so that the register types are also availalbe conditionally (which earlier was the case due to the conditional use above)? e.g. like 

```suggestion
#[derive(Copy, Clone, Eq, PartialEq)] {# Peripheral definition#}
{% set peri_struct = p.name | to_struct_id -%}
#[cfg(feature = "{{module_name}}")]
pub struct {{ peri_struct }}(*mut u8);
#[cfg(feature = "{{module_name}}")]
unsafe impl core::marker::Send for {{ peri_struct }} {}
#[cfg(feature = "{{module_name}}")]
unsafe impl core::marker::Sync for {{ peri_struct }} {}
```

or via a putting them into a module?

It does not generate any "real" issues, but makes a bunch of possibly disabled stuff show up in intellisense.

Reply:
Thanks for suggestions. Regarding the conditional compilation I have implemented in a slightly simpler way